src/main/java/teammates/ui/controller/StudentCourseJoinAction.java

```diff
@@ -34,8 +35,13 @@ public ActionResult execute() throws EntityDoesNotExistException {
         
         String confirmUrl = Const.ActionURIs.STUDENT_COURSE_JOIN_AUTHENTICATED 
                 + "?" + Const.ParamsNames.REGKEY + "=" + regkey 
-                + "&" + Const.ParamsNames.NEXT_URL + "=" + nextUrl;
-        data = new StudentCourseJoinConfirmationPageData(account, student, confirmUrl, Logic.getLogoutUrl(confirmUrl));
+                + "&" + Const.ParamsNames.NEXT_URL + "=" + Sanitizer.sanitizeForNextUrl(nextUrl);
+        String nextUrlType = nextUrl.replaceFirst("^(/page/[A-Za-z]+)\\?.*$", "$1");
```

Comment:
It can be a static method in this class itself?
Whenever we use regex (not just in this case), I'd like to see the behaviour specified in terms of comments and tests because we can't expect all devs to know regex. Extracting it to small static method gives us a chance to do both. A regex has a lot of hidden logic in it that is not even captured by coverage tools. A unit test targeting only this line will give us an opportunity to write a whole lot of cheap unit tests that illustrating the behavior using negative and positive test cases.